lib/utils.ts

```diff
@@ -193,3 +195,23 @@ export const isValidGjsColor = (color: string): boolean => {
 export const capitalizeFirstLetter = (str: string): string => {
     return str.charAt(0).toUpperCase() + str.slice(1);
 };
+
+export function warnOnLowBattery(): void {
+    const { lowBatteryThreshold, lowBatteryNotification } = options.menus.power;
+
+    Utils.merge([lowBatteryThreshold.bind('value'), lowBatteryNotification.bind('value')], (low, enabled) => {
+        if (!enabled) return;
+        battery.connect('notify::percent', () => {
+            if (battery.percent === low || battery.percent === low / 2 || !battery.charging) {
+                Utils.execAsync([
+                    'notify-send',
+                    `${battery.percent}% Battery Percentage`,
+                    '-i',
+                    icons.battery.warning,
+                    '-u',
+                    'critical',
+                ]);
+            }
+        });
+    });
+}
```

Comment:
You shouldn't need the top level bindings here since the values being checked will be at the moment that the event triggers. So if it triggers at 20%, then at that moment it will check the values as they are, so no need to bind them.

```suggestion
export function warnOnLowBattery(): void {
    battery.connect('notify::percent', () => {
        const { lowBatteryThreshold, lowBatteryNotification } = options.menus.power;

        if (!lowBatteryNotification.value || battery.charging) return;

        const lowThreshold = lowBatteryThreshold.value;

        if (battery.percent === lowThreshold || battery.percent === lowThreshold / 2) {
            Utils.execAsync([
                'notify-send',
                `${battery.percent}% Battery Percentage`,
                '-i',
                icons.battery.warning,
                '-u',
                'critical',
            ]);
        }
    });
}
```

Reply:
will this be reactive. When I update the from the settings will it be reflected back on the current ags session or have to restart

Reply:
It will be because every time the battery.connect event triggers, it'll fetch the value at that point on time. All these values are, are just variables at the end of the day.

So if you update the value in the settings, the next time the event triggers after your settings change, it will fetch the variables value which will be the new value.

Reply:
Oh yeah since its tied to event its supposed to be reactive.